src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfileDataFlowExample.cs

```diff
@@ -0,0 +1,134 @@
+/// that. Enter the IProjectDynamicLoadComponent. Implementations of this interface are "loaded" whenever their
+/// capabilities are satisfied, and "unloaded" whenever the capabilities are no longer satisfied. So here we export
+/// this as an IProjectDynamicLoadComponent, and use the AppliesTo attibute to tie it to the LaunchProfiles
+/// capability.
+/// 
+/// Note we specify ExportContractNames.Scopes.UnconfiguredProject in the Export attribute because we expect/need
+/// one instance of this type per UnconfiguredProject.
+/// </remarks>
+[Export(ExportContractNames.Scopes.UnconfiguredProject, typeof(IProjectDynamicLoadComponent))]
+[AppliesTo(ProjectCapabilities.LaunchProfiles)]
+internal class LaunchProfileDataFlowExample : IProjectDynamicLoadComponent
+{
+    /// <remarks>
+    /// We need the UnconfiguredProject for several reasons:
+    ///   1. We need the full path to the project file so we can edit the properties in that file when the launch
+    ///      settings change.
+    ///   2. It provides context for fault handling. If this type throws an exception while processing the data from
+    ///      the data flow, the project system can pop up a gold bar blaming the affected project and letting the
+    ///      user know that things may not work properly.
+    ///   3. Importing an UnconfiguredProject in the constructor also guarantees we get a new instance of this type
+    ///      associated with each UnconfiguredProject (though this is a bit redundant as the other two imports would
+    ///      also cause this).
+    /// </remarks>
+    private readonly UnconfiguredProject _project;
+    /// <remarks>
+    /// We need the ILaunchSettingsProvider in order to hook up to ILaunchSettingsProvider.SourceBlock.
+    /// </remarks>
+    private readonly ILaunchSettingsProvider _launchSettingsProvider;
+    /// <remarks>
+    /// The IProjectPropertiesProvider is needed to update the project properties when the launch settings change.
+    /// </remarks>
+    private readonly IProjectPropertiesProvider _projectProperties;
+
+    /// <remarks>
+    /// Hooking up to a data flow source creates a link; disposing the link is how to break the connection to the data
+    /// flow source when we no longer need it.
+    /// </remarks>
+    private IDisposable? _launchSettingsLink;
+    /// <remarks>
+    /// We need to store some state, namely, the previous command name/debug target.
+    /// </remarks>
+    private string? _previousCommandName = null;
+
+    [ImportingConstructor]
+    public LaunchProfileDataFlowExample(
+        UnconfiguredProject unconfiguredProject,
+        ILaunchSettingsProvider launchSettingsProvider,
+        // There are multiple implementations of IProjectPropertiesProvider. The Import attribute here tells MEF to give
+        // us the specific one named "ProjectFile".
+        [Import(ContractNames.ProjectPropertyProviders.ProjectFile)] IProjectPropertiesProvider projectProperties)
```

Comment:
`ContractNames.ProjectPropertyProviders.ProjectFile` is defined in https://github.com/dotnet/project-system/blob/main/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ContractNames.cs.